test/pkg/controller/namespace/namespace_controller.go

```diff
@@ -0,0 +1,101 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namespace
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+/**
+* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
+* business logic.  Delete these comments after modifying this file.*
+ */
+
+// Add creates a new Namespace Controller and adds it to the Manager.  The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+// USER ACTION REQUIRED: update cmd/manager/main.go to call this core.Add(mrg) to install this Controller
+func Add(mrg manager.Manager) error {
+	return add(mrg, newReconcile(mrg))
+}
+
+// newReconcile returns a new reconcile.Reconcile
+func newReconcile(mrg manager.Manager) reconcile.Reconcile {
+	return &ReconcileNamespace{client: mrg.GetClient()}
+}
+
+// add adds a new Controller to mrg with r as the reconcile.Reconcile
+func add(mrg manager.Manager, r reconcile.Reconcile) error {
+	// Create a new controller
+	c, err := controller.New("namespace-controller", mrg, controller.Options{Reconcile: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to Namespace
+	err = c.Watch(&source.Kind{Type: &corev1.Namespace{}}, &handler.Enqueue{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create
+	// Uncomment watch a Deployment created by Namespace - change this for objects you create
+	err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueOwner{
+		IsController: true,
+		OwnerType:    &corev1.Namespace{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ReconcileNamespace reconciles a Namespace object
+type ReconcileNamespace struct {
+	client client.Client
+}
+
+// Reconcile reads that state of the cluster for a Namespace object and makes changes based on the state read
+// and what is in the Namespace.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  The scaffolding writes
+// a Deployment as an example
+func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the Namespace instance
+	instance := &corev1.Namespace{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
```

Comment:
Is this always true?
What if this reconcile.Request is for a Delete event?

Reply:
Then the object won't be found.  :)